tests/unit/s2n_init_thread_test.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <pthread.h>
+
+#include "s2n_test.h"
+
+static void *s2n_initialize_thread_cb(void *_unused_arg)
+{
+    (void) _unused_arg;
+
+    EXPECT_SUCCESS(s2n_init());
+    return NULL;
+}
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST_NO_INIT();
+
+    /* Tests our thread-local memory cleanup. 
+     *
```

Comment:
Why create a separate test for this? Doesn't s2n_init_test cover this?

Reply:
No it's not the same test. The s2n_init_test checks that you can't have multi-invocations of s2n_init(), even in a different thread. This test is checking that s2n_init() memory allocated in a thread is automatically cleaned up.

I could put the test in the s2n_init_test file 🤔